### PR TITLE
feat: added params validation to missing routes

### PR DESCRIPTION
**Pull request by andreabadesso**
…h joi

### CODE REVIEW

File: src/api/balances.ts [hunk 1]
```diff
@@ -14,25 +14,45 @@ import { getWalletBalances } from '@src/commons';
...
 import { closeDbConnection, getDbConnection, getUnixTimestamp } from '@src/utils';
 
 const mysql = getDbConnection();
 
+const paramsSchema = Joi.object({
+  id: Joi.string()
+    .required(),
+  token_id: Joi.string()
+    .alphanum()
+    .required()
```
Comment (pedroferreira1) [inline src/api/balances.ts]: Why is it required and optional?

File: src/api/txhistory.ts [hunk 2]
```diff
@@ -16,14 +16,32 @@ import {
...
 const MAX_COUNT = 15;
-
 const htrToken = hathorLib.constants.HATHOR_TOKEN_CONFIG.uid;
 
+const paramsSchema = Joi.object({
+  id: Joi.string()
+    .required(),
+  token_id: Joi.string()
+    .alphanum()
+    .required()
```
Comment (pedroferreira1) [inline src/api/txhistory.ts]: It shouldn't be required, right?

File: src/api/balances.ts [hunk 3]
```diff
@@ -14,25 +14,45 @@ import { getWalletBalances } from '@src/commons';
...
+    .optional(),
```
Comment (pedroferreira1) [inline src/api/balances.ts:31]: Should we have the default as HTR? Or the API returns all token balances if no token_id is used?

File: src/api/txhistory.ts [hunk 4]
```diff
@@ -16,14 +16,32 @@ import {
...
+    .default(htrToken)
+    .optional(),
+  skip: Joi.number()
```
Comment (pedroferreira1) [inline src/api/txhistory.ts:32]: Should we add here `min(0)`?

File: src/api/txhistory.ts [hunk 5]
```diff
@@ -16,14 +16,32 @@ import {
...
+    .integer()
+    .default(0)
+    .optional(),
+  count: Joi.number()
+    .integer()
+    .default(MAX_COUNT)
+    .optional(),
```
Comment (pedroferreira1) [inline src/api/txhistory.ts:42]: Maybe add `.positive()` and `max(MAX_COUNT)`

File: src/api/wallet.ts [hunk 6]
```diff
@@ -15,26 +15,39 @@ import {
...
-    walletId = params.id;
-  } else {
-    return {
-      statusCode: 200,
-      body: JSON.stringify({ success: false, error: ApiError.MISSING_PARAMETER, parameter: 'id' }),
-    };
+
+  const { value, error } = getParamsSchema.validate(params, {
+    abortEarly: false,
+    convert: true, // Skip and count will come as query params as strings
```
Comment (pedroferreira1) [inline src/api/wallet.ts]: I think this comment is not from here, right?

**Review by pedroferreira1 — CHANGES_REQUESTED** [review]

File: src/api/balances.ts (hunk 1 above)
Comment (andreabadesso) [reply src/api/balances.ts]: Typo. Fixed

File: src/api/txhistory.ts (hunk 2 above)
Comment (andreabadesso) [reply src/api/txhistory.ts]: Typo. Fixed

File: src/api/txhistory.ts (hunk 4 above)
Comment (andreabadesso) [reply src/api/txhistory.ts:32]: I agree. Done

File: src/api/txhistory.ts (hunk 5 above)
Comment (andreabadesso) [reply src/api/txhistory.ts:42]: I agree. Done

File: src/api/wallet.ts (hunk 6 above)
Comment (andreabadesso) [reply src/api/wallet.ts]: Typo. Fixed

File: src/api/balances.ts (hunk 3 above)
Comment (msbrogli) [reply src/api/balances.ts:31]: Maybe we should add a parameter for that (e.g. `all_tokens`).

File: src/api/txhistory.ts [hunk 7]
```diff
@@ -32,41 +52,27 @@ const htrToken = hathorLib.constants.HATHOR_TOKEN_CONFIG.uid;
 export const get: APIGatewayProxyHandler = async (event) => {
   const params = event.queryStringParameters;
 
-  let walletId: string;
-  if (params && params.id) {
-    walletId = params.id;
-  } else {
-    return closeDbAndGetError(mysql, ApiError.MISSING_PARAMETER, { parameter: 'id' });
-  }
+  const { value, error } = paramsSchema.validate(params, {
```
Comment (msbrogli) [inline src/api/txhistory.ts:55]: This code repeats itself many times. Should we move the options to a constant?

**Review by msbrogli — APPROVED** [review]

File: src/api/balances.ts (hunk 3 above)
Comment (andreabadesso) [reply src/api/balances.ts:31]: By default we return all balances from all tokens if token_id is not passed

I like `token_id` parameter being a filter and suggest keeping all tokens as a default

Comment (pedroferreira1) [reply src/api/balances.ts:31]: I agree to keep returning all tokens by default and only filter when `token_uid` is passed.

**Review by pedroferreira1 — APPROVED** [review]

File: src/api/txhistory.ts (hunk 7 above)
Comment (andreabadesso) [reply src/api/txhistory.ts:55]: The options are not always the same, I want `convert` to be false when the params are coming from the query string

Comment (msbrogli) [reply src/api/txhistory.ts:55]: Ok. I haven't noticed that.

File: src/api/balances.ts (hunk 3 above)
Comment (msbrogli) [reply src/api/balances.ts:31]: Ok. Just add a note that we expect the number of tokens in a wallet to be small.

**Review by msbrogli — APPROVED** [review]

File: src/api/balances.ts (hunk 3 above)
Comment (andreabadesso) [reply src/api/balances.ts:31]: Done in 9834750f23bf9a75db15d8b0a5d0ab2ce4e6df52

**Pull request merged**